node/silkworm/common/log.cpp

```diff
@@ -68,6 +71,10 @@ static inline std::pair<const char*, const char*> get_channel_settings(Level lev
 }
 
 BufferBase::BufferBase(Level level) : level_(level) {
+    if (level > settings_.log_verbosity) {
+        return;
+    }
```

Comment:
This is an improvement for sure. I think the same early return should be added also into the `append` method to avoid unnecessary work, i.e. appending to the `std::stringstream` instance.

Reply:
Will amend ... nevertheless I wish we could focus a bit more on overall design of this PR instead of searching performance bounds for logging which, btw, is really not a critical part (we don't log hundreds times per second).

Reply:
Done